frontend/src/edit.jsx

```diff
@@ -1,21 +1,142 @@
-  const { open } = props
-  console.log('render edit menu', open);
+  useEffect(async () => {
+    if (parcelID) {
+      const table = await getLulcTableForParcel(parcelCoords);
+      setParcelTable(table);
+    }
+  }, [parcelID]);
 
-  if (open) {
-    return (
-      <div className={"menu-container"}>
-        <span>Wallpaper with pre-made></span>
-        {/*<Dropdown>
+  useInterval(async () => {
+    console.log('checking status for', jobID);
+    const status = await getStatus(jobID);
+    if (status === 'complete') {
+      const results = await getWallpaperResults(jobID);
+      setParcelTable(results);
+      setJobID(null);
+    }
+  }, jobID ? 1000 : null);
 
-        </Dropdown>*/}
-      </div>
-    );
-  } else {
-    return <div />
+  async function handleSubmitNew(event) {
+    event.preventDefault();
+    if (!scenarioName) {
+      alert('no scenario was selected');
+      return;
+    }
+    if (!pattern) {
+      alert('no modification options selected; no changes to make');
+      return;
+    }
+    if (!parcelID) {
+      alert('no parcel was selected; no changes to make');
+      return;
+    }
+    let sid = scenarioID;
+    if (!Object.values(savedScenarios).includes(scenarioName)) {
+      sid = await makeScenario(scenarioName, 'description');
+      setScenarioID(sid);
+    }
+    const jid = await doWallpaper(parcelCoords, pattern, sid);
+    setJobID(jid);
+    refreshSavedScenarios();
   }
+
+  // TODO: do handlers  need to be wrapped in useCallback? 
+  // to memoize the function?
+  const handleRadio = (event) => {
+    setPattern(event.target.value);
+  };
+
+  const handleTabChange = (tabID) => {
+    setActiveTab(tabID);
+  };
+
+  return (
+    <div className="menu-container">
+      <Tabs id="Tabs" onChange={handleTabChange} selectedTabId={activeTab}>
+        <Tab
+          id="create"
+          title="Create"
+          panel={(
+            <div>
+              <ParcelTable parcelTable={parcelTable} />
+              <br />
+              <form onSubmit={handleSubmitNew}>
+                <RadioGroup
+                  inline={false}
+                  label="Modify the landuse of this parcel by selecting a pattern:"
+                  onChange={handleRadio}
+                  selectedValue={pattern}
+                >
+                  <Radio
+                    value="orchard"
+                    label="orchard"
+                  />
+                  <Radio
+                    value="city park"
+                    label="city park"
+                  />
+                  <Radio
+                    value="housing"
+                    label="housing"
+                  />
+                </RadioGroup>
+                <h4>Add this modification to a scenario</h4>
+                <datalist id="scenariolist">
+                  {Object.values(savedScenarios).map(item => <option key={item} value={item} />)}
+                </datalist>
+                <input
+                  type="search"
```

Comment:
This is fine for now, but I wonder if there's any prebuilt components for something like an editable dropdown, where you can see the existing options but also type in a new one?

Reply:
I think that's what this does. It has a dropdown that shows all options, or a filtered list of them once you start typing. And it accepts new values too. But I agree we can probably find something better eventually.

Reply:
Ah okay, I missed the dropdown feature.